### PR TITLE
EC-2694 License selection

### PR DESCRIPTION
**Pull request by lbiedinger**

### CODE REVIEW

File: app/helpers/ugc_form_helper.rb [hunk 1]
```diff
@@ -31,4 +31,27 @@ def contribution_state_change_input(builder, events, selected)
   def aasm_events_for_select(events)
     events.map { |event| [t(event.name, scope: 'contribute.contributions.events'), event.name] }
   end
+
+  ##
+  # To look up the mongo documents for:
+  # https://creativecommons.org/publicdomain/mark/1.0/
```
Comment (rwd) [inline app/helpers/ugc_form_helper.rb]: http://, not https://, in this and the next

File: app/helpers/ugc_form_helper.rb [hunk 2]
```diff
@@ -31,4 +31,27 @@ def contribution_state_change_input(builder, events, selected)
...
+  # http://rightsstatements.org/page/CNE/1.0/
+  def edm_rights_options
+    [
+      [edm_rights_label_html('public_domain'), rights_id_from_url('http://creativecommons.org/publicdomain/mark/1.0/')],
+      [edm_rights_label_html('creative_commons_attribution_share_alike'), rights_id_from_url('http://creativecommons.org/licenses/by-sa/4.0/')],
+      [edm_rights_label_html('copyright_not_evaluated'), rights_id_from_url('http://rightsstatements.org/vocab/CNE/1.0/')]
+    ]
+  end
+
+  def rights_id_from_url(rights_url)
```
Comment (rwd) [inline app/helpers/ugc_form_helper.rb]: Could we name this `cc_license_id_from_url`?

File: app/helpers/ugc_form_helper.rb [hunk 3]
```diff
@@ -31,4 +31,27 @@ def contribution_state_change_input(builder, events, selected)
...
   end
+
+  ##
+  # To look up the mongo documents for:
+  # https://creativecommons.org/publicdomain/mark/1.0/
+  # https://creativecommons.org/licenses/by-sa/4.0/
+  # http://rightsstatements.org/page/CNE/1.0/
+  def edm_rights_options
+    [
+      [edm_rights_label_html('public_domain'), rights_id_from_url('http://creativecommons.org/publicdomain/mark/1.0/')],
```
Comment (rwd) [inline app/helpers/ugc_form_helper.rb]: I am reluctant to have to maintain a hard-coded mapping of I18n keys to license URIs. Let's derive the former from the latter instead:
```ruby
def cc_license_i18n_key(rdf_about)
  uri = URI.parse(rdf_about)
  uri.host.tr('.', '_') + uri.path.tr('/.', '._').sub(/\.\z/, '')
end

CC::License.each { |l| puts cc_license_i18n_key(l.rdf_about) }
```
```
creativecommons_org.publicdomain.mark.1_0
rightsstatements_org.vocab.NoC-NC.1_0
rightsstatements_org.vocab.NoC-OKLR.1_0
creativecommons_org.publicdomain.zero.1_0
creativecommons_org.licenses.by.4_0
creativecommons_org.licenses.by-sa.4_0
creativecommons_org.licenses.by-nd.4_0
creativecommons_org.licenses.by-nc.4_0
creativecommons_org.licenses.by-nc-sa.4_0
creativecommons_org.licenses.by-nc-nd.4_0
rightsstatements_org.vocab.InC.1_0
rightsstatements_org.vocab.InC-EDU.1_0
rightsstatements_org.vocab.InC-OW-EU.1_0
rightsstatements_org.vocab.CNE.1_0
```

File: app/models/edm/web_resource.rb [hunk 4]
```diff
@@ -11,8 +11,7 @@ class WebResource
     mount_uploader :media, MediaUploader
 
     belongs_to :edm_rights,
-               class_name: 'CC::License', inverse_of: :edm_rights_for_edm_web_resources,
-               optional: true
+               class_name: 'CC::License', inverse_of: :edm_rights_for_edm_web_resources
```
Comment (rwd) [inline app/models/edm/web_resource.rb]: `edm:rights` is an *optional* property on `edm:WebResource`. This should be reflected in the model, and requirement enforced elsewhere, e.g. in per-campaign validation rules.

File: app/views/migration/_form.html.erb [hunk 5]
```diff
@@ -54,14 +54,15 @@
             <%= wr.input(:dc_type) %>
             <%= wr.input(:dcterms_created, as: :date, html5: true, input_html: date_format_fallback_attributes) %>
             <%= wr.input(:dc_creator) %>
+            <%= wr.input(:edm_rights, as: :radio_buttons, collection: edm_rights_options, checked: wr.object&.edm_rights&.id, wrapper_html: { 'data-error-msg-key': 'contribute.campaigns.migration.form.validation.web-resource-license' }) %>
```
Comment (rwd) [inline app/views/migration/_form.html.erb]: Is it necessary to specify `checked` here?

Comment (lbiedinger) [reply app/views/migration/_form.html.erb]: yes

File: app/models/edm/web_resource.rb [hunk 6]
```diff
@@ -39,6 +40,7 @@ class WebResource
 
     validates :media, presence: true, if: :published?
     validate :europeana_supported_media_mime_type, unless: :media_blank?
+    validates_presence_of :edm_rights, unless: :media_blank?
```
Comment (rwd) [inline app/models/edm/web_resource.rb]: Change to `validates :edm_rights, presence: true, unless: :media_blank?`

File: app/views/migration/_form.html.erb [hunk 7]
```diff
@@ -54,14 +54,15 @@
...
           <% end %>
         <% end %>
 
         <%= agg.nested_fields_for(:edm_hasViews, wrapper_tag: :div) do |wr| %>
           <%= field_set_tag do %>
             <%= wr.remove_nested_fields_link(i18n.t('buttons.object.remove')) %>
-            <%= wr.input(:media, label: i18n.t('legends.each_object', number: ''), label_html: { class: 'sequenced_object' }, hint: i18n.t('hints.edm_hasView.media'), input_html: { accept: "#{EDM::WebResource.allowed_extensions}, #{EDM::WebResource.allowed_content_types}" }) %>
-            <%= wr.input(:media_cache, as: :hidden) %>
+            <%= wr.input(:media, label: i18n.t('legends.each_object', number: ''), label_html: { class: 'sequenced_object' }, hint: i18n.t('hints.edm_hasView.media'), input_html: { accept: "#{EDM::WebResource.allowed_extensions}, #{EDM::WebResource.allowed_content_types}", 'data-makes-required': "contribution_ore_aggregation_edm_hasViews___nested_field_for_replace_with_index___edm_rights" }) %>
```
Comment (rwd) [inline app/views/migration/_form.html.erb]: Single quote the strings for the values of the `data-makes-required` attributes.

File: app/views/migration/_form.html.erb (hunk 5 above)
Comment (rwd) [reply app/views/migration/_form.html.erb]: Change it to `wr.input(:edm_rights_id)`, update the permitted attribute in the controller, and `checked` can be removed, plus fewer MongoDB queries need be made on the edit action.

**Pull request merged**